hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/calculator/ContractCreateFeeCalculator.java

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: Apache-2.0
+package com.hedera.node.app.service.contract.impl.calculator;
+
+import static com.hedera.node.app.spi.fees.SimpleFeeCalculatorImpl.countKeys;
+import static org.hiero.hapi.fees.FeeScheduleUtils.lookupServiceFee;
+import static org.hiero.hapi.support.fees.Extra.KEYS;
+
+import com.hedera.hapi.node.base.HederaFunctionality;
+import com.hedera.hapi.node.transaction.TransactionBody;
+import com.hedera.node.app.spi.fees.FeeContext;
+import com.hedera.node.app.spi.fees.ServiceFeeCalculator;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import org.hiero.hapi.fees.FeeResult;
+import org.hiero.hapi.support.fees.FeeSchedule;
+import org.hiero.hapi.support.fees.ServiceFeeDefinition;
+
+public class ContractCreateFeeCalculator implements ServiceFeeCalculator {
+    @Override
+    public void accumulateServiceFee(
+            @NonNull final TransactionBody txnBody,
+            @Nullable final FeeContext feeContext,
+            @NonNull final FeeResult feeResult,
+            @NonNull final FeeSchedule feeSchedule) {
+        final var adminKey = txnBody.contractCreateInstanceOrThrow().adminKey();
+        final long keys = adminKey != null ? countKeys(adminKey) : 0;
+        final ServiceFeeDefinition serviceDef = lookupServiceFee(feeSchedule, HederaFunctionality.CONTRACT_CREATE);
+        feeResult.addServiceFee(1, serviceDef.baseFee());
+        addExtraFee(feeResult, serviceDef, KEYS, feeSchedule, keys);
```

Comment:
Same here, This fee calculation says we havee included some [gas](https://github.com/hashgraph/hedera-fee-tool-js/blob/master/src/resources/typedUsageFormulas.json#L1497). Should we account for that?